templates/types/streaming/nextjs/app/components/ui/file-content-preview.tsx

```diff
@@ -14,24 +14,76 @@ import {
 } from "./drawer";
 import { cn } from "./lib/utils";
 
-export interface UploadCsvPreviewProps {
-  csv: CsvFile;
+export interface FileContentPreviewProps {
+  file: ContentFile;
   onRemove?: () => void;
 }
 
-export default function UploadCsvPreview(props: UploadCsvPreviewProps) {
-  const { filename, filesize, content } = props.csv;
+export function FileContentPreview({
+  file,
+  onRemove,
+}: FileContentPreviewProps) {
+  if (file.filetype === "csv") {
+    return (
+      <PreviewDialog
+        file={file}
+        title="Csv Raw Content"
+        card={{
+          icon: (
+            <div className="relative h-10 w-10 shrink-0 overflow-hidden rounded-md">
+              <Image
```

Comment:
if we add an image for PDF too, we don't need the explicit markup for each file type